plugins/compress/compress.cc

```diff
@@ -733,6 +737,26 @@ transformable(TSHttpTxn txnp, bool server, HostConfiguration *host_configuration
     return 0;
   }
 
+  // check Partial Object is transformable
+  if (host_configuration->range_request_ctl() == RangeRequestCtrl::NO_COMPRESSION) {
```

Comment:
This block restores code prior to the PR #11975 and add another check of `Content-Range` header in server response. Because `cache_range_request` might remove `Range` header from client request.
https://github.com/apache/trafficserver/pull/11975/files#diff-a50782f349c34e95c6c8643b834aaeba685f244f78a645e85a402bbda9012fb6L680-L689